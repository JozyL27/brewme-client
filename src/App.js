import React from 'react'
import { Route, Switch } from 'react-router-dom'
import Header from './components/Header/header'
import About from './components/about/about'
import NotFoundPage from './routes/NotFoundPage'
import NavBar from './components/NavBar/NavBar'
import PublicOnlyRoute from './components/Utils/PublicOnlyRoute'
import PrivateRoute from './components/Utils/PrivateRoute'
import LoginPage from './routes/LoginPage'
import SearchBeer from './components/searchBeers/searchBeer'
import RandomBeer from './components/randomBeer/randomBeer'
import MyBeers from './components/MyBeers/MyBeers'
import Logo from './components/Logo/Logo'


class App extends React.Component {
  render() {
    return (
      <div className='App'>
        <header className='App_header'>
        <Header />
        </header>
        <Logo />
        <NavBar />
        <main className='App_main'>
          <Switch>
            <Route
            exact
            path={'/'}
            component={About}
            />
            <PrivateRoute
            path={'/search'}
            component={SearchBeer}
            />
            <PrivateRoute
            path={'/myBeers'}
            component={MyBeers}
            />
            <PrivateRoute
            path={'/random'}
            component={RandomBeer}
            />
            <PublicOnlyRoute 
              path={'/login'}
              component={LoginPage}
            />
            <Route
            component={NotFoundPage}
            />
          </Switch>
        </main>
      </div>
    )
  }
}


export default App