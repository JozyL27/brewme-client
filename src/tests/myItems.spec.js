import React from 'react'
import ReactDOM from 'react-dom'
import App from '../App'
import { BrowserRouter } from 'react-router-dom'
import MyItems from '../components/MyBeers/myItems';
import renderer from 'react-test-renderer'


describe(`MyItems component`, () => {
    const testInput = [{
        name: 'beer one',
        abv: 4,
        descript: 'im a beer',
        user_id: 2,
        beer_id: 1
    },
    {
        name: 'beer two',
        abv: 5,
        descript: 'im a beer too',
        user_id: 3,
        beer_id: 2
    }
    ]

    it(`My App renders without crashing`, () => {
        const div = document.createElement('div')

        ReactDOM.render(
        <BrowserRouter>
        <App />
        </BrowserRouter>,
        div)


        ReactDOM.unmountComponentAtNode(div)
    })

    it(`My List Component renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<MyItems brewskis={testInput}/>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it(`renders the  UI as expected`, () => {
        const tree = renderer
            .create(<MyItems brewskis={testInput}/>)
            .toJSON()
            expect(tree).toMatchSnapshot()
    })
})