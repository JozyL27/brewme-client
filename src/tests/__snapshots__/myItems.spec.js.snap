// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyItems component renders the  UI as expected 1`] = `
Array [
  <li
    className="myLi"
  >
     
    beer one
    <div
      className="myAbv"
    >
      ABV: 
      4
    </div>
    <p
      className="searchDescript"
    >
      im a beer
    </p>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      Delete
    </button>
  </li>,
  <li
    className="myLi"
  >
     
    beer two
    <div
      className="myAbv"
    >
      ABV: 
      5
    </div>
    <p
      className="searchDescript"
    >
      im a beer too
    </p>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      Delete
    </button>
  </li>,
]
`;
